pkg/controllers/inventory/inventory.go

```diff
@@ -47,36 +48,46 @@ func InitInventory() *Inventory {
 
 // BuildVpcCache using vpc list from cloud, update vpc cache(with vpcs applicable for the current cloud account).
 func (inventory *Inventory) BuildVpcCache(vpcMap map[string]*runtimev1alpha1.Vpc, namespacedName *types.NamespacedName) error {
+	var numVpcsToAdd, numVpcsToUpdate, numVpcsToDelete int
 	vpcsInCache, _ := inventory.vpcStore.GetByIndex(common.VpcIndexerByAccountNameSpacedName, namespacedName.String())
 
 	// Remove vpcs in vpc cache which are not found in vpc list fetched from cloud.
 	for _, i := range vpcsInCache {
 		vpc := i.(*runtimev1alpha1.Vpc)
 		if _, found := vpcMap[vpc.Status.Id]; !found {
-			inventory.log.V(1).Info("Deleting vpc from vpc cache", "vpc id", vpc.Status.Id, "account",
-				namespacedName.String())
 			if err := inventory.vpcStore.Delete(fmt.Sprintf("%v/%v-%v", vpc.Namespace, vpc.Labels[common.VpcLabelAccountName],
 				vpc.Status.Id)); err != nil {
 				inventory.log.Error(err, "failed to delete vpc from vpc cache", "vpc id", vpc.Status.Id, "account",
 					namespacedName.String())
 			}
+			numVpcsToDelete++
 		}
 	}
 
-	for _, v := range vpcMap {
+	for _, discoveredVpc := range vpcMap {
 		var err error
-		key := fmt.Sprintf("%v/%v-%v", v.Namespace, v.Labels[common.VpcLabelAccountName], v.Status.Id)
-		if _, found, _ := inventory.vpcStore.Get(key); !found {
-			err = inventory.vpcStore.Create(v)
+		key := fmt.Sprintf("%v/%v-%v", discoveredVpc.Namespace, discoveredVpc.Labels[common.VpcLabelAccountName], discoveredVpc.Status.Id)
+		if cachedObj, found, _ := inventory.vpcStore.Get(key); !found {
+			err = inventory.vpcStore.Create(discoveredVpc)
+			numVpcsToAdd++
 		} else {
-			err = inventory.vpcStore.Update(v)
+			cachedVpc := cachedObj.(*runtimev1alpha1.Vpc)
+			if !utils.AreDiscoveredFieldsSameVpcStatus(cachedVpc.Status, discoveredVpc.Status) {
```

Comment:
Doesnt deepequal compare work??

Reply:
It works, I had thought that it wouldn't work for nested objects like slices, I have updated the code. Also, tested.